File: ZPRO/scripts/ZPRO_C/16/02/Prvek.cpp
#include "Prvek.h"

Prvek::Prvek()
{
	set(NULA, nullptr);
}

Prvek::Prvek(T _data, Prvek * _dalsi)
{
	set(_data, _dalsi);
}

void Prvek::setDalsi(Prvek *_dalsi)
{
	dalsi = _dalsi;
}

void Prvek::setData(T _data)
{
	data = _data;
}

T Prvek::getData()
{
	return data;
}

Prvek* Prvek::getDalsi()
{
	return dalsi;
}

void Prvek::set(T _data, Prvek* _dalsi)
{
	setData(_data);
	setDalsi(_dalsi);
}

File: PJ/Lecture7/GenericList/Prvek.java


public class Prvek<T> 
{
    private T data;
    private Prvek další;
    
    public Prvek(T data, Prvek další)
    {
        setDalší(další);
        setData(data);
    }

    public Prvek getDalší() 
    {
        return další;
    }

    public void setDalší(Prvek další) 
    {
        this.další = další;
    }

    public T getData() 
    {
        return data;
    }

    public void setData(T data) 
    {
        this.data = data;
    }
    
}
